src/models/gift_card.rs

```diff
@@ -0,0 +1,36 @@
+//! Model struct for GiftCard type
+
+use serde::{Deserialize, Serialize};
+
+use super::{
+    enums::{GiftCardGANSource, GiftCardStatus, GiftCardType},
+    DateTime, Money,
+};
+
+/// Represents a Square gift card.
+#[derive(Clone, Debug, Default, Deserialize, Eq, PartialEq, Serialize)]
+pub struct GiftCard {
+    /// **Read only** The Square-assigned ID of the gift card.
+    pub id: Option<String>,
+    /// The gift card type.
+    pub r#type: GiftCardType,
+    /// **Read only** The current gift card balance. This balance is always greater than or equal
+    /// to zero.
+    pub balance_money: Option<Money>,
+    /// **Read only** The timestamp when the gift card was created, in RFC 3339 format. In the case
+    /// of a digital gift card, it is the time when you create a card (using the Square Point of
+    /// Sale application, Seller Dashboard, or Gift Cards API).
+    ///
+    /// In the case of a plastic gift card, it is the time when Square associates the card with the
+    /// seller at the time of activation.
+    pub created_at: Option<DateTime>,
```

Comment:
Many of these feel like they shouldn't be options. Is that because of Square?

Reply:
Yes, any read-only fields are optional, because they're not sent on write.